File: src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"
using namespace rclcpp;
using namespace std;

class NumberPublisherNode : public Node 
{
public:
    NumberPublisherNode() : Node("number_publisher")
    {
        this->declare_parameter("publish_number",2);
        this->declare_parameter("frequency",1.0);
        number_ = this->get_parameter("publish_number").as_int();   //type of data: as_type()
        frequency_ = this->get_parameter("frequency").as_double();

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number",10);
        timer_ = this->create_wall_timer(chrono::milliseconds(int (1000.0 / frequency_)),
                                        bind(&NumberPublisherNode::publishNumbers,this));
        RCLCPP_INFO(this->get_logger(),"Number Publisher has been started.");
    }

private:
    void publishNumbers()
    {
        example_interfaces::msg::Int64 counter_;    //constructor
        counter_.data = number_;
        publisher_->publish(counter_);
    }
    Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;        //SharedPtr is important (don't forget it in C++)
    TimerBase::SharedPtr timer_;
    int number_;
    double frequency_;
};

int main(int argc, char **argv)
{
    init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>(); 
    spin(node);
    shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class NumberPubplisherNode(Node): 
    def __init__(self):
        super().__init__("number_publisher") 
        #default value of publish_number is 2
        self.declare_parameter("publish_number", 2)
        self.number_ = self.get_parameter("publish_number").value
        #default value of frequency is 1.0
        self.declare_parameter("frequency" , 1.0)
        self.frequecy_ = self.get_parameter("frequency").value

        self.publishers_ = self.create_publisher(Int64,"number",10)
        self.timers_ = self.create_timer(1.0 / self.frequecy_,self.publishNumbers)
        self.get_logger().info("Number Publisher has been started.")

    def publishNumbers(self):
        msg = Int64()
        msg.data = self.number_
        self.publishers_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = NumberPubplisherNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()